mlflow/sklearn/__init__.py

```diff
@@ -994,3 +986,79 @@ def patched_fit(original, self, *args, **kwargs):
                 safe_patch(
                     FLAVOR_NAME, class_def, func_name, patched_fit, manage_run=True,
                 )
+
+
+def eval_and_log_metrics(*, model, X, y_true, prefix, sample_weight=None):
+    """
+    Computes and logs metrics (and artifacts) for the given model and labeled dataset.
+    The metrics/artifacts mirror what is auto-logged when training a model
+    (see mlflow.sklearn.autolog).
+
+    ** Example **
+
+    .. code-block:: python
+
+        from sklearn.linear_model import LinearRegression
+        import mlflow
+
+        # enable autologging
+        mlflow.sklearn.autolog()
+
+        # prepare training data
+        X = np.array([[1, 1], [1, 2], [2, 2], [2, 3]])
+        y = np.dot(X, np.array([1, 2])) + 3
+
+        # prepare evaluation data
+        X_eval = np.array([[3, 3], [3, 4]])
+        y_eval = np.dot(X_eval, np.array([1,2])) + 3
+
+        # train a model
+        model = LinearRegression()
+        with mlflow.start_run() as run:
+            model.fit(X, y)
+            (metrics, artifacts) = mlflow.sklearn.eval_and_log_metrics(model=model, X=X_eval,
+                                                                       y_true=y_eval, prefix="val_")
+
+
+    Each metric's and artifact's name is prefixed with `prefix`, e.g., in the previous example the
+    metrics and artifacts are named 'val_XXXXX'. Note that training-time metrics are auto-logged
+    as 'training_XXXXX'. Metrics and artifacs are logged under the current run if one exists,
+    otherwise a new run is started and left open.
+
+    Raises an error if:
+    - prefix is empty
+    - model is not an sklearn estimator or does not support the 'predict' method
+
+    :param model: The model to be evaluated.
```

Comment:
Can we move the arguments block above the example code?

Reply:
Sure thing -- I can address this in the next commit along with the previous comments (once we reach a resolution).

Reply:
Done.

Reply:
Thanks!